lib/chef/provider.rb

```diff
@@ -70,6 +72,29 @@ def action_#{name}
       EOM
     end
 
+    # Return the hash of action descriptions defined for
+    # the provider class.
+    #
+    # @return [Hash] hash of [Symbol] => [String] containing
+    # any provided action descriptions.
+    def self.action_descriptions
+      @action_descriptions ||= {}
+    end
+
+    # Retrieve the description for a provider's action, if
+    # any description has been included in the definition.
+    #
+    # @param action [Symbol,String] the action name
+    # @return [String] the description of the action provided, or nil if no description
+    # was defined
+    def self.action_description(action)
+      description = action_descriptions[action.to_sym]
+      if description.nil? && superclass.respond_to?(:action_description)
+        description = superclass.action_description(action)
+      end
```

Comment:
I was gonna say that you should loop over `while description.nil?` and iterate up the superclass tree, but didn't have a good suggestion on how to get the superclass's superclass.

But I saw below the code you used to have to dup the action_description from the parent in the constructor and I kind of thing you should do that instead...

Reply:
This will iterate by calling up the class hierarchy with `superclass.action_description`